### PR TITLE
feat: reorganize the new SDK package

### PR DESCRIPTION
**Pull request by frostming**
Signed-off-by: Frost Ming <me@frostming.com>

## What does this PR address?

Fixes #(issue)

## Before submitting:

- [ ] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
      guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [ ] Does the code follow BentoML's code style, `pre-commit run -a` script has passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [ ] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [ ] Did your changes require updates to the documentation? Have you updated
      those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

### CODE REVIEW

File: src/bentoml/__init__.py [hunk 1]
```diff
@@ -94,10 +95,15 @@
     from . import batch  # Batch API
     from . import exceptions  # BentoML exceptions
     from . import server  # Server API
+    from . import types  # BentoML types
```
[the remark this answers is not in the document]
Comment (frostming) [reply src/bentoml/__init__.py:98]: Thanks, got it, I think the more appropriate place is `bentoml.io` but for now there is name conflicting between the new IO descriptor and old ones.

[the remark this answers is not in the document]
Comment (frostming) [reply src/bentoml/__init__.py]: I've moved those IO descriptors to `bentoml.ui`, not sure if it is a proper name.

File: examples/sentence-embedding/service.py [hunk 2]
```diff
@@ -2,18 +2,14 @@
 
 import os
 
-import bentoml_io
 import numpy as np
 import torch
-from pydantic import Field
 
 import bentoml
+from bentoml.ui import Field
```
Comment (aarnphm) [inline examples/sentence-embedding/service.py:9]: how about `bentoml.io.Field`?

Comment (frostming) [reply examples/sentence-embedding/service.py:9]: But the other members(`File`, `Image` can't be imported from `bentoml.io`)

Comment (aarnphm) [reply examples/sentence-embedding/service.py:9]: How about `bentoml.Field`?

Comment (frostming) [reply examples/sentence-embedding/service.py:9]: Are we going to also move other IO descriptors to the top namespace?

`bentoml.Image`, `bentoml.File`

Comment (aarnphm) [reply examples/sentence-embedding/service.py:9]: @ssheng what are we feeling about this?

I'm fine with make this top level.

**Pull request merged**